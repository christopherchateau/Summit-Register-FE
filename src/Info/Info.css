.Info {
  align-items: center;
  display: flex;
  flex-direction: column;
  background: url("../utilities/Images/mountains-2.jpg") no-repeat center
  rgba(165, 165, 165, 0.68);
  background-size: auto 70%;
  height: 90vh;
  justify-content: space-between;

  border: 4px solid black;
  height: 100vh;
}

.header {
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: space-between;
  padding-top: 12px;
  width: 80%;
}

.back-btn {
  background: transparent;
  border: none;
  display: block;
  font-size: 22px;
  margin: 70px 0;
  padding: 5px;
  transition: all 0.4s ease-in-out;
}

.back-btn:hover {
  background: black;
  color: white;
  transition: all 0.4s ease-in-out;
}

.logo-small {
  height: 50px;
  width: 90px;
}

.info-section {
  border: 1px solid white;
  border-radius: 20px;
  background: grey;
  margin-bottom: 20px;
  margin-top: 50px;
  opacity: .9;
  padding: 20px;
}

.view-log-btn,
.sign-register-btn {
  border: 1px solid black;
  border-radius: 20px;
  display: block;
  font-size: 16px;
  /* padding: 5px 10px; */
}
