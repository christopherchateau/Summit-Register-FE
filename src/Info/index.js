import React, { Component } from "react";
import HourlyForecast from "../HourlyForecast";
import Green from "../utilities/Images/green.png";
import Blue from "../utilities/Images/blue.png";
import Black from "../utilities/Images/black.png";
import DoubleBlack from "../utilities/Images/double_black.png";
import PropTypes from "prop-types";
import "./Info.css";

class Info extends Component {
  constructor() {
    super();
    this.state = {
      display: "info"
    };
  }

  toggleView = () => {
    this.state.display === "info"
      ? this.setState({ display: "weather" })
      : this.setState({ display: "info" });
  };

  render() {
    const {
      currentMountainWeather,
      withinRange,
      handleViewLogButton
    } = this.props;

    const {
      altitude,
      difficulty,
      range
    } = this.props.currentMountainData.attributes;

    const difficultyIcons = {
      Green,
      Blue,
      Black,
      "Double Black": DoubleBlack
    };

    let displayText;
    withinRange
      ? (displayText =
          "Congratulations, you're within range to sign the register!")
      : (displayText =
          "You are not currenlty within range to sign the register.");

    const hourlyForecasts = currentMountainWeather.data
      .slice(0, 12)
      .map(forecast => <HourlyForecast key={forecast.time} {...forecast} />);

    const info = (
      <div>
        <img
          className="difficulty-icon"
          alt={difficulty}
          src={difficultyIcons[difficulty]}
        />
        <h3>Altitude: {altitude} ft</h3>
        <h3>Range: {range}</h3>
        <hr />
        <h3 className="display-text">{displayText}</h3>
      </div>
    );

    const weather = (
      <div className="weather">
        <h3 className="weather-summary-text">
          {currentMountainWeather.summary}
        </h3>
        <section className="hourly-forecast-section">{hourlyForecasts}</section>
      </div>
    );

    return (
      <div className="Info">
        <section className="info-section">
          {this.state.display === "info" ? info : weather}
          <div className="btn-container">
            <button
              className="view-log-btn"
              onClick={() => handleViewLogButton()}
            >
              Log
            </button>
            <button className="weather-btn" onClick={() => this.toggleView()}>
              {this.state.display === "info" ? "Weather" : "Info"}
            </button>
          </div>
        </section>
      </div>
    );
  }
}

Info.propTypes = {
  handleBackButton: PropTypes.func,
  currentMountainData: PropTypes.object,
  handleviewLogButton: PropTypes.func,
  handleLogUpdate: PropTypes.func
};

export default Info;
