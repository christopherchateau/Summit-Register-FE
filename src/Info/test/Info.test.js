configure({ adapter: new Adapter() });
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Info from "..";

describe("Info", () => {
  let wrapper;
  const currentMountainData = {
    attributes: {
      name: "Mt Massive",
      altitude: 999999,
      difficulty: "super hard",
      range: "Front Range"
    }
  };
  beforeEach(() => {
    wrapper = shallow(<Info currentMountainData={currentMountainData} />);
  });

  it.skip("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it.skip("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Info currentMountainData={currentMountainData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it.skip("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
