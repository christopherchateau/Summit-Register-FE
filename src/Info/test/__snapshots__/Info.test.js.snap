// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should match snapshot 1`] = `undefined`;
